src/renderer/services/wallets.js

```diff
@@ -194,6 +578,234 @@ export default {
+
+      if (btcTestnet.err || btcMainnet.err) {
+        reject({
+          err: true,
+        });
+      } else {
+        resolve({
+          key,
+          btcTestnetWalletClient: btcTestnet.walletClient,
+          btcMainnetWalletClient: btcMainnet.walletClient,
+        });
+      }
+    });
+  },
+
+  generateBTCwalletKey() {
+    const Key = this.getKey();
+
+    const lang = {
+      name: 'English',
+      isoCode: 'en',
+    };
+
+    // creating key
+
+    const key = Key.create({
+      lang // eslint-disable-line comma-dangle
+    });
+
+    return ({
+      key,
+    });
+  },
+
+  createTestnetBTCWallet(key, opts) {
+    return new Promise((resolve, reject) => {
+      opts.networkName = 'testnet';
+      const walletClient = this.getClient(null, opts);
+      walletClient.fromString(
+        key.createCredentials('', {
+          coin: opts.coin,
+          network: 'testnet',
+          account: opts.account || 0,
+          n: opts.n || 1 // eslint-disable-line comma-dangle
+        }) // eslint-disable-line comma-dangle
+      );
+
+      walletClient.createWallet(
+        opts.name,
+        opts.myName,
+        opts.m,
+        opts.n,
+        {
+          network: 'testnet',
+          singleAddress: opts.singleAddress,
+          walletPrivKey: opts.walletPrivKey,
+          coin: opts.coin // eslint-disable-line comma-dangle
+        },
+        (err) => {
+          if (err) {
+            console.log(err);
+            reject({
+              err: true,
+            });
+          } else {
+            resolve({
+              walletClient,
+            });
+          }
+        } // eslint-disable-line comma-dangle
+      );
+    });
+  },
+
+  createMainnetBTCWallet(key, opts) {
+    return new Promise((resolve, reject) => {
+      opts.networkName = 'livenet';
+      const walletClient = this.getClient(null, opts);
+      walletClient.fromString(
+        key.createCredentials('', {
+          coin: opts.coin,
+          network: 'livenet',
+          account: opts.account || 0,
+          n: opts.n || 1 // eslint-disable-line comma-dangle
+        }),
+      );
+
+      walletClient.createWallet(
+        opts.name,
+        opts.myName,
+        opts.m,
+        opts.n,
+        {
+          network: 'livenet',
+          singleAddress: opts.singleAddress,
+          walletPrivKey: opts.walletPrivKey,
+          coin: opts.coin // eslint-disable-line comma-dangle
+        },
+        (err) => {
+          if (err) {
+            console.log(err);
+            reject({
+              err: true,
+            });
+          } else {
+            resolve({
+              walletClient,
+            });
+          }
+        } // eslint-disable-line comma-dangle
+      );
+    });
+  },
+
+  getBTCWalletObjects(mnemonic) {
+    return new Promise((resolve, reject) => {
+      const opts = {
+        bwsurl: 'https://bws.bitpay.com/bws/api',
+        passphrase: null,
+        words: mnemonic,
```

Comment:
I'm hoping/assuming that the client serverAssistedImport() method here does not actually send any private key related data to the server. Can you verify what happens with the mnemonic in the serverAssistedImport method? Since this method is in one of the library dependencies, care has to be taken any time this dependency version is changed to ensure it handles these things without introducing a security issue.